### PR TITLE
PositionControl interface mapping update

### PR DESCRIPTION
**Pull request by Stifael**
based on the discussion from [here](https://github.com/PX4/Firmware/issues/10734)
- set integral states and setpoint of reference state to 0
- set thrust to NAN if it will be computed from position and velocity control

### CODE REVIEW

**Comment by RomanBapst** [comment]
@Stifael Thanks! I'll test fly it tomorrow and will report back with results.

File: src/modules/mc_pos_control/PositionControl.cpp [hunk 1]
```diff
@@ -277,7 +263,7 @@ void PositionControl::_velocityController(const float &dt)
 	// Saturate thrust setpoint in D-direction.
 	_thr_sp(2) = math::constrain(thrust_desired_D, uMin, uMax);
 
-	if (fabsf(_thr_sp(0)) + fabsf(_thr_sp(1))  > FLT_EPSILON) {
+	if (PX4_ISFINITE(_thr_sp(0)) + PX4_ISFINITE(_thr_sp(1))) {
```
Comment (RomanBapst) [inline src/modules/mc_pos_control/PositionControl.cpp]: @Stifael This is a bit weird. Can we change this to check if both values are finite?

**Comment by RomanBapst** [comment]
Fixes #10737

File: src/modules/mc_pos_control/PositionControl.cpp (hunk 1 above)
Comment (Stifael) [reply src/modules/mc_pos_control/PositionControl.cpp]: yep, sorry for that.

**Comment by RomanBapst** [comment]
@Stifael Thanks! I just rebased this. Should be good to go. I'll do some SITL testing.

**Pull request merged**